streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1453,7 +1461,9 @@ private Thread shutdownHelper(final boolean error, final long timeoutMs, final b
         return new Thread(() -> {
             // notify all the threads to stop; avoid deadlocks by stopping any
             // further state reports from the thread since we're shutting down
-            int numStreamThreads = processStreamThread(streamThread -> streamThread.shutdown(leaveGroup));
+            int numStreamThreads = processStreamThread(
+                streamThread -> streamThread.shutdown(operation == org.apache.kafka.streams.CloseOptions.GroupMembershipOperation.LEAVE_GROUP)
```

Comment:
I'm thinking we could update the `StreamThread#shutdown` to take a `CloseOptions.GroupMembershipOperation` parameter, then change the `AtomicBoolean` to `AtomicReference<GroupMembershipOperation>` and we can pass the close option directly to the consumer.  WDYT?

Reply:
I think updating the `StreamThread#shutdown` parameter makes sense to me, so I will update the PR.